dsls/sql/src/main/java/org/apache/beam/dsls/sql/interpreter/operator/BeamSqlCastExpression.java

```
@@ -0,0 +1,127 @@
+      case SMALLINT:
+        return BeamSqlPrimitive
+            .of(castOutputType, SqlFunctions.toShort(opValueEvaluated(index, inputRecord)));
+      case TINYINT:
+        return BeamSqlPrimitive
+            .of(castOutputType, SqlFunctions.toByte(opValueEvaluated(index, inputRecord)));
+      case BIGINT:
+        return BeamSqlPrimitive
+            .of(castOutputType, SqlFunctions.toLong(opValueEvaluated(index, inputRecord)));
+      case DECIMAL:
+        return BeamSqlPrimitive
+            .of(castOutputType, SqlFunctions.toBigDecimal(opValueEvaluated(index, inputRecord)));
+      case FLOAT:
+        return BeamSqlPrimitive
+            .of(castOutputType, SqlFunctions.toFloat(opValueEvaluated(index, inputRecord)));
+      case CHAR:
+      case VARCHAR:
+        return BeamSqlPrimitive.of(castOutputType, opValueEvaluated(index, inputRecord).toString());
+      case DATE:
+        return BeamSqlPrimitive
+            .of(castOutputType, toDate(opValueEvaluated(index, inputRecord), outputDateFormat));
+      case TIMESTAMP:
+        return BeamSqlPrimitive.of(castOutputType,
+            toTimeStamp(opValueEvaluated(index, inputRecord), outputTimestampFormat));
+    }
+    throw new RuntimeException(String.format("Cast to type %s not supported", castOutputType));
+  }
+
+  private Date toDate(Object inputDate, String outputFormat) {
+    try {
+      return Date
+          .valueOf(dateTimeFormatter.parseLocalDate(inputDate.toString()).toString(outputFormat));
+    } catch (IllegalArgumentException | UnsupportedOperationException e) {
+      return null;
+    }
+  }
+
+  private Timestamp toTimeStamp(Object inputTimestamp, String outputFormat) {
+    try {
+      return Timestamp.valueOf(
+          dateTimeFormatter.parseDateTime(inputTimestamp.toString()).secondOfMinute()
+              .roundCeilingCopy().toString(outputFormat));
+    } catch (IllegalArgumentException | UnsupportedOperationException e) {
+      return null;
```

throw the exception instead of `return null`